lib/streamlit/DeltaGenerator.py

```diff
@@ -2505,16 +2505,17 @@ def progress(self, element, value):
         ...     my_bar.progress(percent_complete + 1)
 
         """
-        # Needed for python 2/3 compatibility
-        value_type = type(value).__name__
-        if value_type == "float":
+        int_value = isinstance(value, numbers.Integral)
```

Comment:
I'd expect to either see these tests be int/float or numbers.Integral/numbers.Real. Is there a type apart from `int` that would have passed the old test of `type(value).__name__ == "int"` (and that we want to capture here)? If not, I'd favor int/float.

Reply:
We get users passing in values of type numpy.int64, numpy.int32, etc.  You can see this at work in the st.number_input processor.

What I really should do is make utility function shared between these, and keep the explanation there.